File: src-tauri/src/dev/device.rs
#[derive(Clone, Serialize)]
pub struct DeviceInfo {
    pub guid: String,
    pub name: String,
    pub apo_installed: bool,
}

impl DeviceInfo {
    pub fn enumerate() -> Result<Vec<DeviceInfo>, AppError> {
        Ok(vec![DeviceInfo{
            guid: "xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
            name: "Dummy Device",
            apo_installed: true,
        }])
    }
}

File: src/types/device.ts
export type DeviceInfo = {
  name: string,
  guid: string,
  apo_installed: boolean,
  is_default: boolean
};

export function deviceName(info: DeviceInfo) {
  if (info.name === 'all') return info.name;
  return `${info.name.replace(/[()]/g, '')} ${info.guid}`;
}
